posthog/temporal/common/clickhouse.py

```diff
@@ -159,6 +159,13 @@ def __init__(self, query, query_id: str):
         super().__init__(query, f"Query with ID '{query_id}' was not found in query log")
 
 
+class ClickHouseMemoryLimitExceededError(ClickHouseError):
```

Comment:
thought: Maybe we could go harder here and start capturing the specific parameters of the memory limit as attributes in the exception (e.g. the maximum memory, allocation, etc...).

But just a thought for later that our error handling for clickhouse could be a bit better. This is already an improvement.